Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformConstants;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformHelper;
+import gov.hhs.fha.nhinc.transform.audit.XDRTransforms;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import gov.hhs.fha.nhinc.util.HomeCommunityMap;
+import java.io.ByteArrayOutputStream;
+import java.lang.management.ManagementFactory;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse;
+
+/**
+ * Class provides generic Audit Data Transformation APIs utilized across all the services
+ *
+ * @author nsubrama
+ */
+public class CoreX12AuditDataTransform {
+
+    private static final XDRTransforms oXDR = new XDRTransforms();
+
+    private static final Logger LOG = Logger.getLogger(CoreX12AuditDataTransform.class);
+
+    public static AuditSourceIdentificationType getAuditSourceIdentificationType() {
+        String hcid = HomeCommunityMap.getLocalHomeCommunityId();
+        String homeCommunityName = HomeCommunityMap.getHomeCommunityName(hcid);
+        return AuditDataTransformHelper.createAuditSourceIdentification(hcid, homeCommunityName);
+    }
+
+    protected static AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        // Create Active Participant Section
+        // create a method to call the AuditDataTransformHelper - one expectation
+        AuditMessageType.ActiveParticipant participant = createActiveParticipantFromUser(
+            oUserInfo, Boolean.TRUE);
+        if (null != oUserInfo.getRoleCoded()) {
+            participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(oUserInfo.getRoleCoded().getCode(), "",
+                oUserInfo.getRoleCoded().getCodeSystemName(), oUserInfo.getRoleCoded().getDisplayName()));
+        }
+        return participant;
+    }
+
+    /**
+     * Create the <code>AuditMessageType.ActiveParticipant</code> for an audit log record.
+     *
+     * @param userInfo
+     * @param userIsReq
+     * @return <code>AuditMessageType.ActiveParticipant</code>
+     */
+    private static AuditMessageType.ActiveParticipant createActiveParticipantFromUser(UserType userInfo,
+        Boolean userIsReq) {
+        String ipAddr = null;
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        if (ipAddr == null) {
```

Comment:
unnecessary null check.  this value will always be null.  you set it right before here.